lib/ansible/modules/storage/zfs/zpool.py

```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, Remy Mudingay <remy.mudingay[at]esss.se>
+# Copyright: (c) 2018, Stephane Armanet <stephane.armanet[at]esss.se>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: zpool
+short_description: Manage zfs zpools
+version_added: "2.7"
+description:
+  - Manage virtual storage pools using zfs zpools
+options:
+  name:
+    description:
+      - Pool name
+    required: true
+  add:
+    description:
+      - Add devices (spare or mirror to an existing zpool)
+    type: bool
+    default: false
+  raid_level:
+    description:
+      - Pool raid level
+    choices: [ raid0, mirror, raidz, raidz1, raidz2, raidz3 ]
+  vdev:
+    description:
+      - Number of devices in a vdev
+    type: int
+  devices:
+    description:
+      - Full path to list of block devices such as hdd, nvme or file
+  ashift:
+    description:
+      - Alignment shift can be used to improve performance and is set once during the creation
+    choices: [ 0, 9, 10, 11, 12, 13, 14, 15, 16 ]
+    type: int
+    default: 0
+  sets:
+    description:
+      - Set options for existing pools
+    type: bool
+    default: false
+  autoreplace:
+    description:
+      - Automatically replace a bad device in pool using a spare device
+    type: bool
+  autoexpand:
+    description:
+      - Enable or disable automatic pool expansion when a larger disk replaces a smaller disk
+    type: bool
+  spare:
+    description:
+      - Full path to list of block devices such as hdd, nvme or nvme
+  zil:
+    description:
+      - ZFS intent log device or devices when mirrored
+    type: str
+  l2arc:
+    description:
+      - ZFS cache device or devices
+    type: str
+  state:
+    description:
+      - Create or delete the pool
+    choices: [ absent, present ]
+    required: true
+author:
+- Remy Mudingay
```

Comment:
The authors should include a Github handle, like:
```suggestion
- Remy Mudingay (@rmudingay)
```
This is needed so you are properly notified as a maintainer.